src/typeChecker/parseTreeTypes.prelude.ts

```diff
@@ -0,0 +1,77 @@
+export const parseTreeTypesPrelude = `
+
+// Types used to describe the shape of the parse tree
+// that is generated by the parse function in Source 4.
+
+// Program
+type Program = Pair<"sequence", Pair<List<Statement>, null>>;
+
+// Statement
+type Statement =
+  | ConstantDeclaration
+  | VariableDeclaration
+  | FunctionDeclaration
+  | ReturnStatement
+  | ConditionalStatement
+  | WhileLoop
+  | ForLoop
+  | BreakStatement
+  | ContinueStatement
+  | Block
+  | Expression;
+
+type ConstantDeclaration = Pair<"constant_declaration", Pair<Name, Pair<Expression, null>>>;
+type FunctionDeclaration = Pair<"function_declaration", Pair<Name, Pair<Parameters, Pair<Block, null>>>>;
+type ReturnStatement = Pair<"return_statement", Pair<Expression, null>>;
+type WhileLoop = Pair<"while_loop", Pair<Expression, Pair<Block, null>>>;
+type ForLoop = Pair<"for_loop", Pair<Expression | VariableDeclaration, Pair<Expression, Pair<Expression, Pair<Block, null>>>>>;
+type BreakStatement = Pair<"break_statement", null>;
+type ContinueStatement = Pair<"continue_statement", null>;
+
+// Parameters
+type Parameters = List<Name>;
+
+// If-Statement
```

Comment:
Do these comments do anything?

Reply:
The comments used to denote the sections that correspond to the BNF form in the Source 4 specs.